values.yaml

```diff
@@ -5,6 +5,7 @@ replicaCount: 1
 image:
   name: xwiki:lts-mysql-tomcat
   pullPolicy: IfNotPresent
+  tag: 123
```

Comment:
is that a leftover?

Reply:
I was earlier passing this variable through the command line. So to make the CLI command clean I shifted it here. Pushed it in this pull request as it is not a part of the logic.

Reply:
It is chart versioning number, I was keeping at random. 
Should I update it to 0.10-beta?
Which versioning system we should use?

Reply:
Random versioning is one of the worst scheme that I can imagine so definitely not this one! ;)

What does this version represent? Is it tied to the XWiki version? For example for the Docker images we use the tags you can see at https://hub.docker.com/_/xwiki?tab=tags is it similar or different?

Reply:
This is an internal image versioning. This is used to differentiate between XWiki images deployed in Kubernetes if two different xwiki are being used at the same time.  The versioning you are talking about is chart versioning in case of helm(See Charts.yaml file).

Reply:
Sorry by mistake I wrote Chart earlier instead of an image.

Reply:
https://codefresh.io/docs/docs/new-helm/helm-best-practices/#helm-packaging-strategies

Reply:
I quickly read your link and they suggest 2 strategies:
1) Use the same versioning for chart and images (they recommend starting with this actually)
2) Use different versions for chart and images (more complex but interesting if there's a need to version and release them separately).

My guess is that we could do 1) for now and align releases on XWiki version releases (this is what we do for the docker images). In other words, for Docker images, we don't version changes to the image except when we push a new version when a new version of XWiki is out. So you get both the change in the XWiki version + internal changes in the image at the same time. Ofc we would start releasing only when the first chat is ready to be released.

WDYT?

Note: I'm far from an expert though and don't master all the consequences. If you prefer 2, then I'd suggest you start at 1.0-SNAPSHOT and release the first one as 1.0 and the keep increasing to 1.1, etc. until a big change happens, then 2.0, etc. (ie standard semantic versioning).

Reply:
I just checked the community stable helm chart of hadoop and wordpress. They are using appVersioning as tag versioning.
Wordpress link:- 
https://github.com/helm/charts/blob/master/stable/wordpress/values.yaml
https://github.com/helm/charts/blob/master/stable/wordpress/Chart.yaml
Hadoop link:-
https://github.com/helm/charts/blob/master/stable/hadoop/values.yaml
https://github.com/helm/charts/blob/master/stable/hadoop/Chart.yaml


I think we should also use appVersion as by seeing the Kubernetes Dashboard we would know the version of the app we are using.

WDYT?

Reply:
For docker image versioning its typical to use application version because the app version is highly coupled to a specific image, and the same image is generally not reusable for multiple app versions. However, for Helm Charts its typical to specify [separate](https://github.com/helm/charts/blob/30b5c13ae1cad13655c0e3f43c7f4cf525e76944/stable/redis-ha/Chart.yaml#L10) `app` and `chart` versions. This allows using specific chart functionality like support for HA or additional cloud vendors separately from application releases, and multiple docker image (and consequently the app) versions can be used easily using the same chart. This also means we should document and test the chart on various application version currently supported by the community.